jerry-core/parser/js/js-scanner-util.c

```diff
@@ -1588,36 +1596,31 @@ scanner_is_context_needed (parser_context_t *context_p) /**< context */
 #else /* !ENABLED (JERRY_ES2015) */
     JERRY_ASSERT (type == SCANNER_STREAM_TYPE_VAR);
 #endif /* ENABLED (JERRY_ES2015) */
+#endif /* !JERRY_NDEBUG */
 
-    size_t length;
-
-    if (!(data_p[0] & SCANNER_STREAM_UINT16_DIFF))
+    if (!(data & SCANNER_STREAM_UINT16_DIFF))
```

Comment:
Would it make sense to extract common parts? like:
```c
data_p += 2 + 1;
if (!(data & SCANNER_STREAM_UINT16_DIFF)))
{
  if (data_p[2] == 0)
  {
    data_p += sizeof (...);
  }
}
else
{
  data_p += 1;
}
```

Another question? Why "2" and "1" ? is there any info about these constants somewhere?

Reply:
I see this addition is also present in other parts. Maybe we could extact them into a single function to have them only at one place.

Reply:
There is a long comment how the stream is compressed:
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/parser/js/js-scanner.h#L114

I have been thinking about mergeng these functions, but I have doubts. They are similar, but they are also different. I don't know what is the best option here.

Reply:
What if we only create a function to calculate the required bytes to add? eg.:
```c
size_t scanner_stream_next_offset(uint8_t *data_p)
{
  size_t offset = 2 + 1;
  if (!(data_p[0] & SCANNER_STREAM_UINT16_DIFF)))
  {
    if (data_p[2] == 0)
    {
      offset += sizeof (...);
    }
  }
  else
  {
    offset+= 1;
  }
  return offset;
}
```

Reply:
I decided to do the merge. I does not seems that bad, and this is not a hot code.